source/tests/system/nixnet_lin_driver_api_tests.cpp

```diff
@@ -0,0 +1,167 @@
+#include <gmock/gmock.h>
+#include <google/protobuf/util/time_util.h>
+#include <gtest/gtest.h>
+#undef interface
+#include <nixnet/nixnet_client.h>
+
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include <string>
+#include <vector>
+
+#include "device_server.h"
+#include "enumerate_devices.h"
+#include "nixnet_utilities.h"
+
+using namespace nixnet_grpc;
+namespace client = nixnet_grpc::experimental::client;
+namespace pb = google::protobuf;
+using namespace ::testing;
+using namespace ::nixnet_utilities;
+using nlohmann::json;
+
+namespace ni {
+namespace tests {
+namespace system {
+namespace {
+
+typedef pb::uint8 u8;
+typedef pb::uint32 u32;
+typedef pb::uint64 u64;
+typedef double f64;
+
+class NiXnetLINDriverApiTests : public ::testing::Test {
+ protected:
+  NiXnetLINDriverApiTests()
+      : device_server_(DeviceServerInterface::Singleton()),
+        stub_(NiXnet::NewStub(device_server_->InProcessChannel()))
+  {
+    device_server_->ResetServer();
+  }
+  virtual ~NiXnetLINDriverApiTests() {}
+
+  void TearDown() override
+  {
+    device_server_->ResetServer();
+  }
+
+  template <typename TService>
+  std::unique_ptr<typename TService::Stub> create_stub()
+  {
+    return TService::NewStub(device_server_->InProcessChannel());
+  }
+
+  std::unique_ptr<NiXnet::Stub>& stub()
+  {
+    return stub_;
+  }
+
+ private:
+  DeviceServerInterface* device_server_;
+  std::unique_ptr<NiXnet::Stub> stub_;
+};
+
+TEST_F(NiXnetLINDriverApiTests, ConvertFramesToFromSignalsFromExample_FetchData_DataLooksReasonable)
+{
+  constexpr auto NUM_FRAMES = 2;
+  constexpr auto NUM_SIGNALS = 2;
+  std::vector<nixnet_grpc::FrameBufferRequest> frames;
+  auto session = EXPECT_SUCCESS(client::create_session(stub(), "NIXNET_exampleLDF", "Cluster", "SlaveSignal1_U16,MasterSignal1_U16", "", CREATE_SESSION_MODE_SIGNAL_CONVERSION_SINGLE_POINT)).session();
+  nixnet_grpc::FrameRequest* frame = new nixnet_grpc::FrameRequest();
+  frame->set_timestamp(0);
+  frame->add_flags(FrameFlags::FRAME_FLAGS_UNSPECIFIED);
+  frame->set_identifier(4);
+  frame->set_type(FrameType::FRAME_TYPE_LIN_DATA);
+  frame->set_payload("\0\1\2\3\4\5\6\7");
```

Comment:
Does this actually set the whole string as a payload? I think the `\0` in a char* constant will end up null terminating it.

In other cases I have used the `""s` suffix from `std::string_literals`. (A `std::string` can contain null, but a C string can't).

Reply:
Yep, me and Sumit were debugging similar scenario earlier. This sets payload to empty string. Another `protoc` generated workaround we found was to specify size of string explicitly. So below snippet should work too.

`frame->set_payload("\0\1\2\3\4\5\6\7", 8);`